draft-ietf-quic-version-negotiation.md

```diff
@@ -221,18 +248,19 @@ below (using the notation from the "Notational Conventions" section of
 
 ~~~
 Handshake Version Information {
-  Currently Attempted Version (32),
+  Chosen Version (32),
   Other Versions (32) ...,
 }
 ~~~
 {: #fig-hvi-format title="Handshake Version Information Format"}
 
 The content of each field is described below:
 
-Currently Attempted Version:
+Chosen Version:
 
-: This field SHALL be equal to the value of the `Version` field in the long
-header that carries this data.
+: The version that the sender has chosen to use for this connection. This field
+MUST be equal to the value of the Version field in the long header that carries
+this data.
```

Comment:
As I said before, I think that you can - at best - say that the field is generally equal to the Version field in the packet that carries the data.  You can't use "MUST".

Reply:
So the server doesn't need to validate what the client provides (it is picking from the list, and **use** >> **validation**).

The client verifies that (a) the version it sent prior to receiving a Version Negotiation packet is not listed, and (b) it would not have selected a different version if it knew what versions the server truly supports.  Note here that the client's choice is based on what the server signals in other versions, the chosen version (if we drop ordering, as we should).

I realize now that the server can't drop compatible versions from this list.  There's a long and complicated argument for why that doesn't work if you are interested, but I need to eat a well-overdue lunch and it might be better over a video link because it's tricky.

Reply:
> So the server doesn't need to validate what the client provides (it is picking from the list, and **use** >> **validation**).

I've tweaked the text to clarify that this is the version used, not the one from long headers. That allowed me to drop this validation.

> The client verifies that (a) the version it sent prior to receiving a Version Negotiation packet is not listed, and (b) it would not have selected a different version if it knew what versions the server truly supports.

Oh good catch I was missing (b).

> Note here that the client's choice is based on what the server signals in other versions, the chosen version (if we drop ordering, as we should).

Note that in the current proposal we have ordering of the client's versions but not of the server's versions.

> I realize now that the server can't drop compatible versions from this list. There's a long and complicated argument for why that doesn't work if you are interested, but I need to eat a well-overdue lunch and it might be better over a video link because it's tricky.

In the current proposal endpoints can disagree on which versions are compatible so we need them in the list in case the client doesn't think they are compatible.

Reply:
That is not the reason I had, but as the server generates its list after the client, it would know what the client thinks is compatible when generating its list.

Reply:
Oh that's a good point. I'd love to hear the full story over VC one of these days then, but no rush.

But overall, do you think this PR is ready to be merged? (If you do I'll email the list to get WG input)